File: IntroduccionPOO/Ejercicio9/Computadora.java
package IntroduccionPOO.Ejercicio9;

public class Computadora {
    String procesador;
    String memoriaRam;
    String almacenamiento;
    String tarjetaGrafica;
    String placaMadre;
    String fuentePoder;
    boolean encendida;

    public Computadora() {
        this.procesador = "Intel Core i5";   
        this.memoriaRam = "8GB";           
        this.almacenamiento = "500GB SSD";  
        this.tarjetaGrafica = "NVIDIA GeForce GTX 1650"; 
        this.placaMadre = "ASUS Prime B450"; 
        this.fuentePoder = "500W";        
        this.encendida = false;
    }

    public void mostrarComponentes() {
        System.out.println("Componentes de la computadora:");
        System.out.println("  Procesador: " + this.procesador);
        System.out.println("  Memoria RAM: " + this.memoriaRam);
        System.out.println("  Almacenamiento: " + this.almacenamiento);
        System.out.println("  Tarjeta Grafica: " + this.tarjetaGrafica);
        System.out.println("  Placa Madre: " + this.placaMadre);
        System.out.println("  Fuente de Poder: " + this.fuentePoder);
    }


    public void mostrarEstado() {
        if (this.encendida) {
            System.out.println("La computadora esta encendida");
        } else {
            System.out.println("La computadora esta apagada");
        }
    }

    public void encender() {
        if (!this.encendida) {
            System.out.println("Encendiendo la computadora...");
            this.encendida = true;
        } else {
            System.out.println("La computadora ya esta encendida");
        }
    }

    public void apagar() {
        if (this.encendida) {
            System.out.println("Apagando la computadora...");
            this.encendida = false;
        } else {
            System.out.println("La computadora ya esta apagada");
        }
    }

    public static void main(String[] args) {
        
        Computadora miComputadora = new Computadora();

        miComputadora.mostrarComponentes();
        miComputadora.mostrarEstado();

        miComputadora.encender();
        miComputadora.mostrarEstado();

        miComputadora.apagar();
        miComputadora.mostrarEstado();

        miComputadora.apagar(); 
        miComputadora.encender(); 
    }
}

File: IntroduccionPOO/Ejercicio9/Computadora.py
class Computadora:
    def __init__(self):
        self.procesador = "Intel Core i5"  
        self.memoria_ram = "8GB"          
        self.almacenamiento = "500GB SSD" 
        self.tarjeta_grafica = "NVIDIA GeForce GTX 1650" 
        self.placa_madre = "ASUS Prime B450"
        self.fuente_poder = "500W"        
        self.encendida = False

    def mostrar_componentes(self):
        print("Componentes de la computadora:")
        print(f"  Procesador: {self.procesador}")
        print(f"  Memoria RAM: {self.memoria_ram}")
        print(f"  Almacenamiento: {self.almacenamiento}")
        print(f"  Tarjeta Gráfica: {self.tarjeta_grafica}")
        print(f"  Placa Madre: {self.placa_madre}")
        print(f"  Fuente de Poder: {self.fuente_poder}")

    def mostrar_estado(self):
        if self.encendida:
            print("La computadora esta encendida")
        else:
            print("La computadora esta apagada")

    def encender(self):
        if not self.encendida:
            print("Encendiendo la computadora...")
            self.encendida = True
        else:
            print("La computadora ya esta encendida")

    def apagar(self):
        if self.encendida:
            print("Apagando la computadora...")
            self.encendida = False
        else:
            print("La computadora ya esta apagada")


mi_computadora = Computadora()

mi_computadora.mostrar_componentes()
mi_computadora.mostrar_estado()

mi_computadora.encender()
mi_computadora.mostrar_estado()

mi_computadora.apagar()
mi_computadora.mostrar_estado()

mi_computadora.apagar() 
mi_computadora.encender() 